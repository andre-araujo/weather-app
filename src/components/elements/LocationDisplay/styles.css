.container {
  background: rgba(255, 255, 255, 0.8);
  padding: 10px 8px;
}

.title {
  font-weight: bold;
}
