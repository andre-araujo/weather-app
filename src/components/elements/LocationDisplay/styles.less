@import '../../../styles/settings/colors.less';

.container {
  background: rgba(255, 255, 255, 0.8);
  padding: 15px 8px;
  box-shadow: 0 2px 2px rgba(0, 0, 0, .3);
  display: flex;
  align-items: center;
}

.title {
  font-weight: 600;
  font-size: 30px;
  letter-spacing: 0.034em;
  color: @grey;
}

.icon {
  font-size: 50px;
  margin-right: 16px;
  color: @grey;
}
