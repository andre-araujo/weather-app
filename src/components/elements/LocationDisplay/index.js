import React from 'react';
import { string } from 'prop-types';

import Icon from '../Icon';
import styles from './styles.less';

function LocationDisplay({
  city,
  state,
}) {
  return (
    <header className={styles.container}>
      <span className={styles.icon}><Icon.Compass className={styles.icon} /></span>
      <h1 className={styles.title}>
        {city}, {state}
      </h1>
    </header>
  );
}

LocationDisplay.defaultProps = {
  city: '',
  state: '',
};

LocationDisplay.propTypes = {
  city: string,
  state: string,
};

export default LocationDisplay;
