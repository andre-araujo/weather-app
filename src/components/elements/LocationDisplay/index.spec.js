import React from 'react';
import { shallow } from 'enzyme';

import LocationDisplay from './index';

describe('LocationDisplay component', () => {
  it('should match snapshot with text as ,', () => {
    expect(shallow(<LocationDisplay />)).toMatchSnapshot();
  });

  it('should match snapshot with text as city,', () => {
    expect(shallow(<LocationDisplay city="city" />)).toMatchSnapshot();
  });

  it('should match snapshot with text as city, state', () => {
    expect(shallow(<LocationDisplay city="city" state="state" />)).toMatchSnapshot();
  });
});
