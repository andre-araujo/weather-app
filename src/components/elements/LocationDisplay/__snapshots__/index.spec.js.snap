// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDisplay component should match snapshot with text as , 1`] = `
<header>
  <span>
    <Component />
  </span>
  <h1>
    , 
  </h1>
</header>
`;

exports[`LocationDisplay component should match snapshot with text as city, 1`] = `
<header>
  <span>
    <Component />
  </span>
  <h1>
    city
    , 
  </h1>
</header>
`;

exports[`LocationDisplay component should match snapshot with text as city, state 1`] = `
<header>
  <span>
    <Component />
  </span>
  <h1>
    city
    , 
    state
  </h1>
</header>
`;
